File: js/직사각형 네개의 합집합의 면적 구하기(2699).js
let matrix = Array.from({length:100}, () => Array(100).fill(0));
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

for (let i=0; i<4; i++) {
    let [x1, y1, x2, y2] = input[i].split(' ').map(Number);
    for (let a=y1; a<y2; a++) {
        for (let b=x1; b<x2; b++) {
            matrix[a][b] = 1;
        }
    }
}

let result = 0;

for (let col of matrix) {
    result += col.reduce((acc,cur) => acc+cur, 0);
}

console.log(result);

File: python/직사각형 네개의 합집합의 면적 구하기(2699).py
import sys
input = sys.stdin.readline

matrix = [[0] * 100 for _ in range(100)]

for _ in range(4):
    x1, y1, x2, y2 = tuple(map(int, input().split()))
    for i in range(y1,y2):
        for j in range(x1,x2):
            matrix[i][j] = 1

result = 0
for k in matrix:
    result += sum(k)

print(result)